src/Bicep.Core/TypeSystem/Radius/V3/CommonBindings.cs

```diff
@@ -149,15 +149,35 @@ public static class CommonBindings
             Type = new ThreePartType("redislabs.com", "Redis", RadiusResources.CategoryBinding),
             Properties =
             {
-                new TypeProperty("host", LanguageConstants.String, TypePropertyFlags.ReadOnly),
+                new TypeProperty("host", LanguageConstants.String, TypePropertyFlags.None),
                 new TypeProperty("port", LanguageConstants.Int, TypePropertyFlags.None),
-                new TypeProperty("username", LanguageConstants.String, TypePropertyFlags.ReadOnly),
+                new TypeProperty("username", LanguageConstants.String, TypePropertyFlags.None),
+                // The secrets section allows usage of binding expression to specify
+                // custom secrets.
+                //
+                // TODO: It is slightly confusing to set `redis.secrets.connectionString`, and read
+                //       from `redis.connectionString`. However, it is backward compatible with the
+                //       existing model. This will be a point we need to discuss when
+                //       check-pointing the end-to-end for RedisComponent.
+                new TypeProperty("secrets", new ObjectType(
+                                     "secrets",
+                                     validationFlags: TypeSymbolValidationFlags.WarnOnTypeMismatch,
+                                     properties: new []
+                                     {
+                                         new TypeProperty("connectionString", LanguageConstants.String, TypePropertyFlags.None),
```

Comment:
Use `TypePropertyFlags` to make this and password WRITE-only

Reply:
Done